### PR TITLE
Set hostname to host

### PR DESCRIPTION
**Pull request by antonnyman**
https://github.com/nodejs/undici/issues/463

### CODE REVIEW

**Review by ronag — APPROVED** [review]

File: lib/core/client.js [hunk 1]
```diff
@@ -159,7 +159,7 @@ class Client extends EventEmitter {
       }
     }
     this[kTLSSession] = null
-    this[kHostHeader] = `host: ${this[kUrl].hostname}\r\n`
+    this[kHostHeader] = `host: ${this[kUrl].host}\r\n`
```
Comment (ronag) [inline lib/core/client.js:162]: const host = this[kUrl].host ? this[kUrl].host : `${this[kUrl].hostname}${this[kUrl].port ? `:${this[kUrl].port}`}`

**Review by ronag — CHANGES_REQUESTED** [review]

File: lib/core/client.js (hunk 1 above)
Comment (ronag) [reply lib/core/client.js:162]: Also probably an invalid arugment check in constructor to ensure that either host or hostname is set and a string

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by ronag** [comment]
Landed with some minor compat fixes in 908be7db7781579ae327a144509a483a8b28f728

**Pull request closed without merging**